astropy/cosmology/realizations.py

```diff
@@ -17,12 +17,11 @@
 # Pre-defined cosmologies. This loops over the parameter sets in the
 # parameters module and creates a LambdaCDM or FlatLambdaCDM instance
 # with the same name as the parameter set in the current module's namespace.
-# Note this assumes all the cosmologies in parameters are LambdaCDM,
-# which is true at least as of this writing.
 
 for key in parameters.available:
     par = getattr(parameters, key)
-    if par["flat"]:
+
+    if par["cosmology"] == "FlatLambdaCDM":
```

Comment:
Does this need a test?

Reply:
Sure! ... of what sort? like that all our built-in ones are "FlatLambdaCDM"? I think there's currently an implicit test because the suite should fail on the raised warning if cosmology != FlatLambdaCDM.

Reply:
Just a little surprised that this change didn't result in any test updates.

Reply:
Ahh. currently all of them are FlatLambdaCDM and the code was doing a sanity check with the parameter "flat". Specifying the cosmology obviates that need.
Here, or in a followup, I would like to do a consistency check on ALL the key-value pairs in the parameter dictionary, because not everything is being used. That would be a good place to check that for cosmology=FlatLambdaCDM, flat is either True or not present (because not needed).

Reply:
I do think a test is good... I just don't know what I would test ATM! 😆

Reply:
@adrn... bringing it on! 😆